src/schema.graphql

```diff
@@ -91,6 +91,7 @@ type OnChain {
     address: String! @pattern(regexp: "^[A-Za-z0-9]+$", policy: THROW)
     amount: Int! @range(min: 0, policy: THROW)
     memo: String @stringLength(max: 1024, policy: THROW) # TODO: add limit on the mobile app
+    sendAll: Boolean
```

Comment:
I think we could have a cleaner API here by splitting the pay function in 2:
```
 pay(
    address: String! @pattern(regexp: "^[A-Za-z0-9]+$", policy: THROW)
    amount: Int! @range(min: 0, policy: THROW)
    memo: String @stringLength(max: 1024, policy: THROW) # TODO: add limit on the mobile app
): Success

payAll(
    address: String! @pattern(regexp: "^[A-Za-z0-9]+$", policy: THROW)
    memo: String @stringLength(max: 1024, policy: THROW) # TODO: add limit on the mobile app
): Success
```

The reason is that it would be harder to misuse the graphql API. either you pay with an amount, or payAll and no amount is needed.

We would need to add a resolver payAll that would pass the `{ amount: 0, sendAll: true}` to the pay function in the backend, but otherwise nothing would change.

What do you think?

Reply:
I agree, I'll work on it.